lib/posix/syscall.h

```diff
@@ -0,0 +1,27 @@
+#ifndef LIB_POSIX_SYSCALL_H_
+#define LIB_POSIX_SYSCALL_H_
+
+#include <unistd.h>
+
+#include "lib/error/return_if_error.h"
+#include "lib/error/status_or.h"
+#include "lib/posix/errno.h"
+
+namespace posix {
+
+// Trivial syscall wrapper. Checks for error conditions and converts errno to
+// error::Status.
+//
+// Return the result of syscall() on success or an error status on failure.
+//
+// See 'man 2 syscall' for details.
+template <typename... ArgsT>
+inline error::StatusOr<int> Syscall(const int number, ArgsT&&... args) {
+  const auto rv = ::syscall(number, std::forward<ArgsT>(args)...);
```

Comment:
If std::forward doesn't serve a purpose here, then it should probably be removed in the interests of clarity.

Reply:
Consider the case of order of operations:
if I write &array[index] or 1 + 2 * 9 I'm requiring that the reader remember some rules. If I write &(array[index]) or 1 + (2 * 9) the reader only needs to know one rule.

Likewise for std::forward. When universal references are present (or variadic templates in general) I prefer to apply std::forward unconditionally. The a reader/writer needs to know one rule: "use std::forward". If it is use only some of the time, the the reader needs to reason about when to apply the rule.

Just like parenthesis, there is no runtime cost to std::forward and it is always safe to apply. I don't see a compelling reason to increase the amount of information that a reader must know or consider.

Reply:
To the reader, this isn't the difference between 1 + 2 * 9 and 1 + (2 * 9), but rather the difference between the expression being present at all or not. I take your point from the perspective of the writer though. ;)

Reply:
There's another consideration that I didn't mention above. It is somewhat academic in this scenario but I like to apply rules uniformly when possible. Consider the case that ::syscall is actually C++ code and that someday its function signature changes such that std::forward<> is now relevant. I'd rather not make assumptions that complicate or reduce the benefit of future refactors (again this is a general consideration not relevant to ::syscall).